src/Orchard.Web/Modules/Orchard.Fields/Drivers/BooleanFieldDriver.cs

```diff
@@ -35,7 +35,7 @@ public class BooleanFieldDriver : ContentFieldDriver<BooleanField> {
 
         protected override DriverResult Editor(ContentPart part, BooleanField field, dynamic shapeHelper) {
             // if the content item is new, assign the default value
-            if(!part.HasDraft() && !part.HasPublished()) {
+            if(!field.Value.HasValue) {
```

Comment:
Didn't seen the details of cloning, but, as i understand, we can't use the previous test because we work on a content item that is being cloning, and therefore has a draft or is published...

So, after a 1st look, and because i think this method is called only during a regular web request, your code looks good for me. Maybe we need a null check on the action string?

Best